src/components/ProductCard.jsx

```diff
@@ -1,25 +1,28 @@
+import { useState } from "react";
 export default function ProductCard({ product }) {
-  let currentImageIndex = 0;
-  let itemsInCart = 0;
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
+  const [showDescription, setShowDescription] = useState(false);
+  const [itemsInCart, setItemsInCart] = useState(1);
 
   const handleAddToCartClick = () => {
-    itemsInCart++;
-    alert(`you added ${itemsInCart}`);
+    setItemsInCart((itemsInCart)=> itemsInCart+1);
+    alert(`You have ${itemsInCart} products added to your cart`);
   };
+
   return (
     <>
       <div id="image-carousel">
         <img
           src={product.imageUrls[currentImageIndex] + " " + product.name}
           alt={product.name}
         />
-        <button>Next</button>
-        <button>Previous</button>
+        <button onClick={()=>{setCurrentImageIndex((currentImageIndex) => currentImageIndex+1)}} disabled={currentImageIndex == product.imageUrls.length-1}>Next</button>
+        <button onClick={()=>{setCurrentImageIndex((currentImageIndex) => currentImageIndex-1)}} disabled={currentImageIndex == 0}>Previous</button>
       </div>
 
       <h3>{product.name}</h3>
-      <p>{product.description}</p>
-      <button>Show Description</button>
+      {showDescription && <p>{product.description}</p>}
+      <button onClick={()=>{setShowDescription((showDescription)=> !showDescription)}}>{showDescription ? "Hide Description" : "Show Description"}</button>
```

Comment:
No need to to pass a function to ` setShowDescription`. Change it to pass in a value instead